core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -498,6 +505,49 @@ protected Optional<Expression> translateInPredicate(ConnectorExpression value, C
             return Optional.empty();
         }
 
+        protected Optional<Expression> translateSimpleCaseWhen(List<ConnectorExpression> arguments)
+        {
+            int argumentsSize = arguments.size();
+            if (argumentsSize < 3) {
+                return Optional.empty();
+            }
+
+            Optional<Expression> operand = translate(arguments.get(0));
+            if (operand.isEmpty()) {
+                return Optional.empty();
+            }
+
+            ImmutableList.Builder<WhenClause> builder = ImmutableList.builder();
+            for (int i = 1; i < argumentsSize - 1; i += 2) {
+                Optional<WhenClause> whenClause = translateWhenClause(arguments.get(i), arguments.get(i + 1));
+                if (whenClause.isPresent()) {
+                    builder.add(whenClause.get());
+                }
+                else {
+                    return Optional.empty();
+                }
```

Comment:
You can flip condition here to return earlier:
```suggestion
               if (whenClause.isEmpty()) {
           	      return Optional.empty();
               }
                
               builder.add(whenClause.get());
```